x/evm/keeper/state_transition.go

```diff
@@ -416,6 +416,15 @@ func (k *Keeper) ApplyMessageWithConfig(ctx sdk.Context, msg core.Message, trace
 		}
 	}
 
+	// calculate a minimum amount of gas to be charged to sender if GasLimit
+	// is considerably higher than GasUsed to stay more aligned with Tendermint gas mechanics
+	// for more info https://github.com/tharsis/ethermint/issues/1085
+	// NOTE: MinGasDenominator can not be negative as it is validated on ValidateParams
+	decGasLimit := sdk.NewDec(int64(msg.Gas()))
```

Comment:
Can we omit the type in the var name? Or does this change the logic? 

```
	gasLimit := sdk.NewDec(int64(msg.Gas()))
	minimumGasUsed := gasLimit.Mul(cfg.Params.MinGasDenominator)
	gasUsed = sdk.NewDec(int64(gasUsed))
	gasUsed = sdk.MaxDec(minimumGasUsed, gasUsed).RoundInt().Uint64()
```

Reply:
yeah this does change the logic, it is not possible to assigned a `sdk.Dec` to a `uint64` value (what `gasUsed` is).
I do think you had a point so updated the logic to avoid that ugly renaming